pkg/apis/core/v1/conversion.go

```diff
@@ -539,3 +546,82 @@ func Convert_core_PersistentVolumeSpec_To_v1_PersistentVolumeSpec(in *core.Persi
 func Convert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in *v1.PersistentVolumeSpec, out *core.PersistentVolumeSpec, s conversion.Scope) error {
 	return autoConvert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in, out, s)
 }
+
+// HandleSeccompVersionSkew implements the version skew behavior described in:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md#version-skew-strategy
+func HandleSeccompVersionSkew(in *core.Pod, out *v1.Pod) error {
+	// Handle the pod
+	annotationValue := in.Annotations[v1.SeccompPodAnnotationKey]
+	var podInField *core.SeccompProfile
+	if in.Spec.SecurityContext != nil {
+		podInField = in.Spec.SecurityContext.SeccompProfile
+	}
+
+	_, err := syncSeccompAnnotationAndFieldForPod(
+		v1.SeccompPodAnnotationKey, annotationValue, podInField, out,
+	)
+	if err != nil {
+		return errors.Wrapf(err, "synchronizing annotations and security context field for pod: %+v", in)
+	}
+
+	// Handle the containers of the pod
+	for _, ctr := range in.Spec.Containers {
+		var ctrInField *core.SeccompProfile
+		if ctr.SecurityContext != nil {
+			ctrInField = ctr.SecurityContext.SeccompProfile
+		}
+
+		// Find the annotation value
+		annotationKey := v1.SeccompContainerAnnotationKeyPrefix + ctr.Name
+		var annotationValue string
+		for key, value := range in.Annotations {
+			if key == annotationKey {
+				annotationValue = value
+				break
+			}
+		}
+
+		_, err := syncSeccompAnnotationAndFieldForPod(
+			annotationKey, annotationValue, ctrInField, out,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "synchronizing annotations and security context field for container: %+v", ctr)
+		}
+	}
+
+	return nil
+}
+
+func syncSeccompAnnotationAndFieldForPod(annotationKey, annotationValue string, field *core.SeccompProfile, pod *v1.Pod) (*v1.SeccompProfile, error) {
```

Comment:
Let's rename this function and change its signature:
```suggestion
func validateSeccompAnnotationAndProfile(annotationKey, annotationValue string, field *core.SeccompProfile, pod *v1.Pod) (error) {
```
This way I can add a new function later on without having the need for changing this one again.

Reply:
That makes sense, just amended that. :+1: